### PR TITLE
pkg/trace/api: fix container tagging over UDS

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?

This commit correctly checks for the new connection type and pulls the UDS credentials out of it.

### Motivation

We introduced a new connection type in #17917 which caused our special case for UDS to fail, resulting in the loss of container tags for users using UDS container tagging on cgroup v2 hosts.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/api/listener.go [hunk 1]
```diff
@@ -112,16 +113,17 @@ func (ln *measuredListener) Accept() (net.Conn, error) {
 }
 
 // Close implements net.Listener.
-func (ln measuredListener) Close() error {
+func (ln *measuredListener) Close() error {
 	err := ln.Listener.Close()
 	ln.flushMetrics()
-	ln.exit <- struct{}{}
```
Comment (knusbaum) [inline pkg/trace/api/listener.go:118]: This is a minor bug exposed by the test, where a double-close will panic.

File: pkg/trace/api/container_linux_test.go [hunk 2]
```diff
@@ -51,6 +51,7 @@ func TestConnContext(t *testing.T) {
 	if err := os.Chmod(sockPath, 0o722); err != nil {
 		t.Fatalf("error setting socket permissions: %v", err)
 	}
+	ln = NewMeasuredListener(ln, "uds_connections", 10)
```
Comment (ajgajg1134) [inline pkg/trace/api/container_linux_test.go:54]: Does this addition make this test fail without you fixes in the other files?

Comment (knusbaum) [reply pkg/trace/api/container_linux_test.go:54]: Yes it does, and that's exactly the purpose.

**Review by ajgajg1134 — APPROVED** [review]
maybe wanna add a skip-qa label since this won't need it it seems?

**Comment by knusbaum** [comment]
I'm going to leave the QA since someone else can verify this works.

File: releasenotes/notes/fix-uds-tagging-ca55006e201cb283.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    APM: fix a regression in the Trace Agent, causing container tagging
+    with UDS and cgroup v2 to fail
```
Comment (urseberry) [inline releasenotes/notes/fix-uds-tagging-ca55006e201cb283.yaml:12]:
```suggestion
    APM: fix a regression in the Trace Agent that caused container tagging
    with UDS and cgroup v2 to fail
```

**Review by urseberry — APPROVED** [review]
Minor clarification suggestion

**Pull request merged**